### PR TITLE
feat: update `Tagging` model

### PR DESCRIPTION
**Pull request by LuisMartinez15**
EX-4985

Update `Tagging` model.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?
- [x] `Main`

## How has this been tested?

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [x] I have **commented** my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [x] My changes generate **no new warnings**.
- [x] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: packages/search-types/src/schemas/result.schema.ts [hunk 1]
```diff
@@ -24,10 +24,10 @@ export const ResultSchema: Result = {
     value: expect.nullOr(Number)
   },
   tagging: {
-    query: TaggingSchema,
-    click: TaggingSchema,
-    add2cart: TaggingSchema,
-    checkout: TaggingSchema
+    query: TaggingInfoSchema,
```
Comment (javieri-empathy) [inline packages/search-types/src/schemas/result.schema.ts:27]: I've created a [task](https://searchbroker.atlassian.net/browse/EX-5038) to remove this, as it shouldn't be necessary anymore.

File: packages/search-types/src/tagging.model.ts [hunk 2]
```diff
@@ -1,12 +1,40 @@
...
  */
 export interface Tagging {
+  /** {@link TaggingInfo | add to cart tagging}. */
+  add2cart?: TaggingInfo;
+  /** {@link TaggingInfo | checkout tagging}. */
+  checkout?: TaggingInfo;
+  /** {@link TaggingInfo | click tagging}. */
+  click?: TaggingInfo;
+  /** {@link TaggingInfo | query tagging}. */
+  query?: TaggingInfo;
```
Comment (javieri-empathy) [inline packages/search-types/src/tagging.model.ts:24]: Would it be worth to also add the `wishlisht` by default?

**Review by javieri-empathy — DISMISSED** [review]

File: packages/search-types/src/tagging.model.ts (hunk 2 above)
Comment (pmareke) [reply packages/search-types/src/tagging.model.ts:24]: Is ok for me.

File: packages/search-types/src/tagging.model.ts [hunk 3]
```diff
@@ -1,12 +1,40 @@
...
+ * actions (query, click, show...) by making a request with these info.
+ *
+ * @public
+ */
+export interface TaggingInfo {
   /** Tagging URL. */
   url: string;
   /** Params of the tagging URL. */
-  params: Record<string, any>;
+  params: Record<string, string | number | boolean>;
```
Comment (pmareke) [inline packages/search-types/src/tagging.model.ts:39]: I'm not sure if number is a type that we're going to have in the params.

image

File: packages/x-components/src/__stubs__/results-stubs.factory.ts [hunk 4]
```diff
@@ -52,16 +52,16 @@ export function createResultStub(name: string, result?: Partial<Result>): Result
 }
 
 /**
- * Function to create a {@link @empathyco/x-types#ResultTagging | result tagging} mocked object
+ * Function to create a {@link @empathyco/x-types#TaggingInfo | result tagging} mocked object
```
Comment (pmareke) [inline packages/x-components/src/__stubs__/results-stubs.factory.ts:55]: This is `Tagging` instead of `TaggingInfo` right?

**Review by pmareke — CHANGES_REQUESTED** [review]
Well done!

File: packages/x-components/src/__stubs__/results-stubs.factory.ts (hunk 4 above)
Comment (LuisMartinez15) [reply packages/x-components/src/__stubs__/results-stubs.factory.ts:55]: Yep, my bad.

**Review by pmareke — APPROVED** [review]

**Review by javieri-empathy — APPROVED** [review]

**Pull request merged**